modules/andes-core/broker/src/main/java/org/wso2/andes/server/AMQChannel.java

```diff
@@ -385,77 +395,71 @@ private void deliverCurrentMessageIfComplete()
                 }
                 else
                 {
-                    if(destinationQueues == null || _currentMessage.getDestinationQueues().isEmpty())
-                    {
-                        if (_currentMessage.isMandatory() || _currentMessage.isImmediate())
-                        {
-                            if(!AndesContext.getInstance().isClusteringEnabled()) {
-                                _transaction.addPostTransactionAction(new WriteReturnAction(AMQConstant.NO_ROUTE, "No Route for message", _currentMessage));
-                            }
-                        }
-                        else
-                        {
-                            //no need to sync whole topic bindings across cluster. Thus this log has no sense in clustered mode
-                            if(!AndesContext.getInstance().isClusteringEnabled()) {
-                                _logger.warn("MESSAGE DISCARDED: No routes for message - " + createAMQMessage(_currentMessage));
-                            }
+                    // If the destination of the andes channel is not set, set the destination to the routing key to the
+                    // receive message
+                    if (null == andesChannel.getDestination()) {
+                        andesChannel.setDestination(this._currentMessage.getRoutingKey());
+                    }
+
+                    if ((null == destinationQueues || _currentMessage.getDestinationQueues().isEmpty())
```

Comment:
Its more readable if you separate two statements inside if condition with parenthesis.